### PR TITLE
How to debug/troubleshoot

### LINKED ISSUE #536: Needed: A troubleshooting section
Following https://github.com/datalad/datalad/issues/4651#issuecomment-649180205. 
This can fit right into the "How to get help section". It should entail:

- short overview of the different log levels
- how to trigger "debug" messages and how to interpret them (DataLad: ``-l/--log-level`` or env ``$DATALAD_LOG_LEVEL``, git-annex: ``-v/--verbose``, ``--debug``/)
- how to enter a debugging session: ``-dbg``

... what else?

### PR DESCRIPTION
**Pull request by adswa**
Fixes #536.
Feedback welcome. At the moment, this is an overview of logging and a hint on how to get into the debugger, and a collection of funny comics to make reading the section less dreadful :wink:
Also, I found https://github.com/datalad/datalad/issues/4661 while at it.

### CODE REVIEW

**Comment by glatard** [comment]
Hi @adswa, this would be very useful!

Here is a bit of text that you might consider adapting and adding somewhere to this section (not sure what the right place is):

# File doesn't get downloaded by `datalad get <file_name>`, no error message is returned

It is common for `datalad get` errors to originate in `git-annex`, the software used by DataLad to transfer data. Here are a few suggestions to debug them:
* Take a deep breath, or preferably a walk in a nice park :)
* Check that you are using a recent version of `git-annex`
    - `git-annex version` returns the version of `git-annex` on the first line of its input
    - The version number contains the release date of the version in use. For instance, `git-annex version: 8.20200330-g971791563` was released on 30 March 2020.
    - If the version that you are using is older than a few months, consider updating using the instructions [here](http://handbook.datalad.org/en/latest/intro/installation.html)
* Try to download the file using `git-annex get -v -d <file_name>`
   - If this doesn't succeed, the DataLad command may not succeed.
   - Options `-d` and `-v` are here to provide as much verbosity in error messages as possible
* Read the output of `git-annex`, identify the error, breathe again, and solve the issue!
* Examples of possible issues:

|  `git-annex` error       |  A solution that worked once     | 
| :------------- | -----------: |
| `Last exception was: Could not find a suitable TLS CA certificate bundle, invalid path: /etc/pki/tls/certs/ca-bundle.crt [adapters.py:cert_verify:227]` | unset environment variable `CURL_CA_BUNDLE` |
| Permission denied when writing file | Download to a writeable file system

**Comment by adswa** [comment]
That's fantastic @glatard, thank you!

**Comment by adswa** [comment]
I've added your suggestion to this PR - thanks! I might integegrate the "common error messages and what they mean" section into the table format. I like it a lot!

**Comment by adswa** [comment]
ah, and before I forget: @all-contributors please add @glatard for content

**Comment by adswa** [comment]
@mih, can you take a look at this if you get a chance? not urgent

File: docs/basics/101-135-help.rst [hunk 1]
```diff
@@ -118,6 +118,164 @@ The "submit a question link" on `DataLad's GitHub page <https://github.com/datal
...
+
+   - **critical**: Only catastrophes are reported. Currently, there is nothing inside of DataLad that would log at this level, so setting the log level to *critical* will result in getting no details at all, not even about errors or failures.
+   - **error**: With this log level you will receive reports on any errors that occurred within the program during command execution.
+   - **warning**: At this log level, the command execution will report on usual situations and anything that *might* be a problem, in addition to report anything from the *error* log level. This is the default log level.
+   - **info**: This log level will include reports by the program that indicate normal behavior and serve to keep you up to date about the current state of things, in additions to warning and error logging messages.
+   - **debug**: This log level is the most useful to troubleshoot a problem, and results in DataLad telling you *absolutely everything* it possibly can.
+
+   By default, the log level is set to ``warning``. Raising the log level (e.g, to ``error``) will decrease the amount of information and output you will receive, while lowering it (e.g., to ``info``) will increase it.
```
[the remark this answers is not in the document]
Comment (adswa) [reply docs/basics/101-135-help.rst:172]: This assumption is based on https://github.com/datalad/datalad/blob/dd09e4dfd7014438305daf398b833f9e03e03273/datalad/cmdline/common_args.py#L41, but I may be mistaken:

```python
_log_level_names = ['critical', 'error', 'warning', 'info', 'debug']
log_level = (
    'log-level', ('-l', '--log-level'),
    dict(action=LogLevelAction,
         choices=_log_level_names + [str(x) for x in range(1, 10)],
         metavar="LEVEL",
         default='warning',
         help="""set logging verbosity level.  Choose among %s.  Also you can
         specify an integer <10 to provide even more debugging information"""
              % ', '.join(_log_level_names))
)

```

File: docs/basics/101-135-help.rst [hunk 2]
```diff
@@ -118,6 +118,164 @@ The "submit a question link" on `DataLad's GitHub page <https://github.com/datal
...
+
+.. findoutmore:: Logging granularity
+
+   Other than log *levels*, you can also adjust the amount of information provided with numerical granularity. Instead of specifying a log level, provide an integer <10, with higher values denoting more debugging information.
```
[the remark this answers is not in the document]
Comment (adswa) [reply docs/basics/101-135-help.rst]: done

File: docs/basics/101-135-help.rst [hunk 3]
```diff
@@ -118,6 +118,164 @@ The "submit a question link" on `DataLad's GitHub page <https://github.com/datal
...
+   The following log levels can be chosen from:
+
+   - **critical**: Only catastrophes are reported. Currently, there is nothing inside of DataLad that would log at this level, so setting the log level to *critical* will result in getting no details at all, not even about errors or failures.
+   - **error**: With this log level you will receive reports on any errors that occurred within the program during command execution.
+   - **warning**: At this log level, the command execution will report on usual situations and anything that *might* be a problem, in addition to report anything from the *error* log level. This is the default log level.
+   - **info**: This log level will include reports by the program that indicate normal behavior and serve to keep you up to date about the current state of things, in additions to warning and error logging messages.
+   - **debug**: This log level is the most useful to troubleshoot a problem, and results in DataLad telling you *absolutely everything* it possibly can.
```
[the remark this answers is not in the document]
Comment (adswa) [reply docs/basics/101-135-help.rst]: thanks!

**Comment by adswa** [comment]
I'll merge this, I have removed any mention of default log levels.

**Pull request merged**